services/venice-controller/src/test/java/com/linkedin/venice/controller/TestUnusedValueSchemaCleanupService.java

```diff
@@ -0,0 +1,87 @@
+    doReturn(Utils.getUniqueString()).when(store).getName();
+    doReturn(1).when(store).getCurrentVersion();
+    List<Version> versionList = new ArrayList<>();
+    Version v = mock(Version.class);
+    doReturn(null).when(v).getHybridStoreConfig();
+    versionList.add(v);
+    doReturn(versionList).when(store).getVersions();
+
+    return store;
+  }
+
+  @Test
+  public void testCleanupUnusedSchema() throws Exception {
+    VeniceHelixAdmin admin = mock(VeniceHelixAdmin.class);
+    VeniceParentHelixAdmin parentHelixAdmin = mock(VeniceParentHelixAdmin.class);
+    VeniceControllerMultiClusterConfig config = mock(VeniceControllerMultiClusterConfig.class);
+    doReturn(1).when(config).getUnusedSchemaCleanupIntervalMinutes();
+    doReturn(1).when(config).getMinSchemaCountToKeep();
+    VeniceControllerConfig controllerConfig = mock(VeniceControllerConfig.class);
+    doReturn(controllerConfig).when(config).getControllerConfig(any());
+    doReturn(true).when(controllerConfig).isUnusedValueSchemaCleanupServiceEnabled();
+    doReturn(true).when(admin).isLeaderControllerFor(any());
+    Store store = mockStore();
+
+    String clusterName = "test_cluster";
+    Set<String> clusters = new HashSet<>();
+    HelixVeniceClusterResources helixVeniceClusterResources = mock(HelixVeniceClusterResources.class);
+    ReadWriteSchemaRepository schemaRepository = mock(ReadWriteSchemaRepository.class);
+    clusters.add(clusterName);
+    doReturn(clusters).when(config).getClusters();
+    List<Store> storeList = new ArrayList<>();
+    storeList.add(store);
+    doReturn(storeList).when(admin).getAllStores(any());
+    doReturn(helixVeniceClusterResources).when(admin).getHelixVeniceClusterResources(anyString());
+    doReturn(schemaRepository).when(helixVeniceClusterResources).getSchemaRepository();
+    SchemaEntry schemaEntry = new SchemaEntry(4, SCHEMA);
+    doReturn(schemaEntry).when(schemaRepository).getSupersetOrLatestValueSchema(anyString());
+    Collection<SchemaEntry> schemaEntries = new ArrayList<>();
+    schemaEntries.add(new SchemaEntry(1, SCHEMA));
+    schemaEntries.add(new SchemaEntry(2, SCHEMA));
+    schemaEntries.add(new SchemaEntry(3, SCHEMA));
+    schemaEntries.add(new SchemaEntry(4, SCHEMA));
+
+    doReturn(schemaEntries).when(parentHelixAdmin).getValueSchemas(anyString(), anyString());
+    Set<Integer> schemaIds = new HashSet<>();
+    schemaIds.add(3);
+    schemaIds.add(4);
+
+    doReturn(schemaIds).when(parentHelixAdmin).getInUseValueSchemaIds(anyString(), anyString());
+
+    UnusedValueSchemaCleanupService service = new UnusedValueSchemaCleanupService(config, admin, parentHelixAdmin);
+
+    service.startInner();
+    Set<Integer> unusedSchemas = new HashSet<>();
+    unusedSchemas.add(1);
+    unusedSchemas.add(2);
+    TestUtils.waitForNonDeterministicAssertion(
+        1,
+        TimeUnit.SECONDS,
+        () -> verify(admin, atLeast(1)).deleteValueSchemas(clusterName, store.getName(), unusedSchemas));
```

Comment:
I think we should know exactly how many schemas will be deleted so can we use times(x) instead of atLeast(1)?